From 39efef47b659fa952a9ac73f948688ebfbc0d5de Mon Sep 17 00:00:00 2001
From: Stephen Gallagher <sgallagh@redhat.com>
Date: Mon, 18 Apr 2011 11:11:13 -0400
Subject: [PATCH] Always generate kpasswdinfo file

Previously, we only generated it when performing a password change,
but this didn't play nicely with kpasswd.
---
 src/providers/krb5/krb5_auth.c |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/src/providers/krb5/krb5_auth.c b/src/providers/krb5/krb5_auth.c
index 379a97be6446bdf0fc5c9bc120c561a532d454cf..55a06a511688d6c9511a4f68db47046a4bfdc737 100644
--- a/src/providers/krb5/krb5_auth.c
+++ b/src/providers/krb5/krb5_auth.c
@@ -523,8 +523,7 @@ static void krb5_resolve_kdc_done(struct tevent_req *subreq)
         be_mark_offline(state->be_ctx);
         kr->is_offline = true;
     } else {
-        if (state->pd->cmd == SSS_PAM_CHAUTHTOK &&
-            kr->krb5_ctx->kpasswd_service != NULL) {
+        if (kr->krb5_ctx->kpasswd_service != NULL) {
             subreq = be_resolve_server_send(state, state->ev, state->be_ctx,
                                             kr->krb5_ctx->kpasswd_service->name);
             if (subreq == NULL) {
-- 
1.7.4.4

